app/callbacks/lib/callbacks.ts

```diff
@@ -0,0 +1,207 @@
+/* eslint-disable @typescript-eslint/no-use-before-define */
+import { Meteor } from 'meteor/meteor';
+import { Random } from 'meteor/random';
+import _ from 'underscore';
+
+let timed = false;
+
+if (Meteor.isClient) {
+	// eslint-disable-next-line @typescript-eslint/no-var-requires
+	const { getConfig } = require('../../ui-utils/client/config');
+	timed = [getConfig('debug'), getConfig('timed-callbacks')].includes('true');
+}
+type ArgumentTypes<F extends Function> = F extends (...args: infer A) => any ? A : never;
```

Comment:
TypeScript should have a built-in `Parameters<T extends Function` utility type that does exactly this. IIRC it should be available in the version we're running in Rocket.Chat 🤔